data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java

```diff
@@ -469,11 +483,17 @@ public Builder withIsmPolicyFile(final String ismPolicyFile) {
         }
 
         public Builder withAction(final String action) {
-            checkArgument(EnumUtils.isValidEnumIgnoreCase(BulkAction.class, action), "action must be one of the following: " +  BulkAction.values());
+            // Removed validation because action may have expresions
             this.action = action;
             return this;
         }
```

Comment:
What will be the user experience if the action that is passed is invalid? Can we at least validate that it is either a valid expression with (https://github.com/opensearch-project/data-prepper/blob/d3a027a738d65dfcb576884aad895d40f56319ac/data-prepper-api/src/main/java/org/opensearch/dataprepper/expression/ExpressionEvaluator.java#L36) or one of the enum values

Reply:
I am doing it at the run time after evaluating the expression. See 
```
     if (BulkAction.fromOptionValue(eventAction) == null) {
        LOG.error("Unknown action {}, skipping the event", eventAction);     
        invalidActionErrorsCounter.increment();                              
        continue;                                                            
      }    
```
The `isValidExpressionStatement` may not be accurate if the key field passed as `/xyz` is seen as valid expression but there is no `xyz` field. Do you think it is still a good idea to do config time validation?

Reply:
`/xyz`, `xyz`, etc. is not a valid expression. But even if it were it's still better to have the validation. we should do config time validation to check the following

```
private boolean isValidAction() {
   return isValidExpression(action) || isValidEnum(action);
}
```